### ISSUE TITLE
mismatch between sample linkage image and actual linkage file content? looking for matching details

### ISSUE THREAD

**Issue by aleks-mariusz** [BINCHI1990/Link-LR-PPD-and-Domestic-EPCs#1]
first off, amazing project and thank you for sharing these ideas.  I am trying to do something similar, however I am using python and pandas dataframes instead of R, so am trying to re-create your linkage between ppd and epc datasets (using the latest data)

the issue i have is, while it's well documented in the pdf file, the actual screenshot of the linkage data
https://github.com/Bin-Chi/Link-LR-PPD-and-Domestic-EPCs/blob/master/Images/screenshot_of_linkage_code.png
is not visible in the file posted on github

https://github.com/Bin-Chi/Link-LR-PPD-and-Domestic-EPCs/blob/master/PPD_EPC_linkage.R

I was hoping to be able to have the text easily accessible to try to replicate this task (again, using python and pandas) but as it's only described in the pdf, i could really use some actual sample code so i don't have to re-type the 150+ new fields/matching rules please :-)

**Comment by Bin-Chi** [comment]
If you read the readme file, you will find the detailed codes are deposited in UK Data Service with a  license. The code part is under embargo in UK Data Service as it need to wait for the related manuscript to get published in UCL first.

I am sorry I cannot open the code for the public now, but you are able to access the code in UK Data Service within 28 days.

Bin
________________________________
From: aleks ***@***.***>
Sent: 06 May 2021 23:33
To: Bin-Chi/Link-LR-PPD-and-Domestic-EPCs ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [Bin-Chi/Link-LR-PPD-and-Domestic-EPCs] mismatch between sample linkage image and actual linkage file content? looking for matching details (#1)


first off, amazing project and thank you for sharing these ideas. I am trying to do something similar, however I am using python and pandas dataframes instead of R, so am trying to re-create your linkage between ppd and epc datasets..

the issue i have is, while it's well documented in the pdf file, the actual screenshot of the linkage data
https://github.com/Bin-Chi/Link-LR-PPD-and-Domestic-EPCs/blob/master/Images/screenshot_of_linkage_code.png
is not visible in the file posted on github

https://github.com/Bin-Chi/Link-LR-PPD-and-Domestic-EPCs/blob/master/PPD_EPC_linkage.R

I was hoping to be able to have the text easily accessible to try to replicate this task (again, using python and pandas) but as it's only described in the pdf, i could really use some actual sample code so i don't have to re-type the 150+ new fields/matching rules please :-)

**Comment by aleks-mariusz** [comment]
Ahh ok sorry, i see that part of the readme now - i am not very familiar with the processes for these research papers so apologies i didn't realize this.  but thank you for replying so quickly..  are the set of codes/matching rules much different than what's available in the working paper series? they contain a description of the approach but it would be a lot of work to go through each data field manipulation, perform it , and implement each individual the matching rules.. (the reason for my original request).

**Comment by Bin-Chi** [comment]
The data linkage method used here is similar to an earlier published method in CASA Working paper 213, but with greater granularity in the matching rules.

Bin
________________________________

**Comment by qianyiloh** [comment]
> first off, amazing project and thank you for sharing these ideas. I am trying to do something similar, however I am using python and pandas dataframes instead of R, so am trying to re-create your linkage between ppd and epc datasets (using the latest data)
> 
> the issue i have is, while it's well documented in the pdf file, the actual screenshot of the linkage data https://github.com/Bin-Chi/Link-LR-PPD-and-Domestic-EPCs/blob/master/Images/screenshot_of_linkage_code.png is not visible in the file posted on github
> 
> https://github.com/Bin-Chi/Link-LR-PPD-and-Domestic-EPCs/blob/master/PPD_EPC_linkage.R
> 
> I was hoping to be able to have the text easily accessible to try to replicate this task (again, using python and pandas) but as it's only described in the pdf, i could really use some actual sample code so i don't have to re-type the 150+ new fields/matching rules please :-)

@aleks-mariusz  wondering whether you’d be open to sharing how you managed to replicate the script in Python as I am attempting the same thing with some difficulty

**Comment by aleks-mariusz** [comment]
@qianyiloh Hey, it's been a while since I looked at this (two years), so i had to dig through my old files and figure out what i did.  What difficulty are you facing specifically?

Looks like I actually didn't get that far myself where i correlated the two datasets and merged them into one, but what i had ended up doing was creating two python notebooks, one that loads the epc certificates and one that lods the land registry data.. both notebooks just create a sqlite database (for easy querying and being able to process the data "offline" (after the script restarts).

I also had a third notebook that i used to process the data (looking up the place i used to live).

I uploaded them [here](https://gist.github.com/aleks-mariusz/5aad8d669c809a671c24010771664861), maybe this can be some help to you ? If you haven't used python (jupyter) notebooks, you are missing out :-) great for explorative "pythoning" especially in the data science arena but really for anything (i use them as a crude IDE when i'm throwing together scripts too).
